src/box/lua/config/applier/box_cfg.lua

```diff
@@ -283,7 +283,9 @@ local function names_apply(config, missing_names, schema_version)
     names_alert_missing(config, missing_names)
     -- Don't wait for box.status to change, we may be already rw, set names
     -- on reload, if it's possible and needed.
-    names_try_set_missing()
+    if schema_version and schema_version >= mkversion.get_latest() then
```

Comment:
Actually, it can. The user may run `box.schema.upgrade()` before upgrading all the instances to the latest tarantool version. Most probably something will break, if the upgrade is non-trivial. But there is a possibility that it'll succeed.

Besides, the user may simply do a replace to space `_schema` with a fake huge version. Again, something will break in this case, but we shouldn't assert that the version is never greater than `latest_version`.

TBH I think we're lacking a knob to prohibit DDL, when schema version is greater than the last known to this tarantool build. Similarly to how we prohibit DDL when schema version is too old.

Long story short, I'd keep this check as is. This will be consistent with other DDL, which wouldn't error if real schema version is greater than the last_version

Reply:
I'm ok with that, thank you for the explanation